### ISSUE TITLE
we have conflicts in our parser compiling

### ISSUE THREAD

**Issue by romange** [dragonflydb/dragonfly#1779]
```
1/7] Generating parser from parser.y
parser.y: warning: 9 shift/reduce conflicts [-Wconflicts-sr]
parser.y: note: rerun with option '-Wcounterexamples' to generate conflict counterexamples

```
https://github.com/dragonflydb/dragonfly/blob/main/src/core/search/parser.y
to reproduce run `ninja query_parser`

**Comment by romange** [comment]
to test `ninja search_parser_test search_test`   and then run both tests to validate the parcing logic

**Comment by kostasrim** [comment]
@dranikpg I think I saw this on one of your PR's. I might be wrong but I am pinging you here

**Comment by romange** [comment]
Please do not work on it @dranikpg

**Comment by rohan-puri** [comment]
When i build this using ```ninja dragonfly```, i get the following conflicts,
```
rohan@Ubuntu-22:~/open-source/dragonfly/build-dbg$ ninja dragonfly
[412/511] Generating parser from parser.y
parser.y: warning: 9 shift/reduce conflicts [-Wconflicts-sr]
parser.y: note: rerun with option '-Wcounterexamples' to generate conflict counterexamples
[511/511] Linking CXX executable dragonfly
```
But when i run the commands @romange you mentioned, no errors show up.
```
rohan@Ubuntu-22:~/open-source/dragonfly/build-dbg$ ninja search_parser_test
[28/28] Linking CXX executable search_parser_test
rohan@Ubuntu-22:~/open-source/dragonfly/build-dbg$ ninja search_parser_test search_test
[2/2] Linking CXX executable search_test
```

**Comment by rohan-puri** [comment]
the way you get more information about this is,
```
rohan@Ubuntu-22:~/open-source/dragonfly/src/core/search$ bison parser.y -Wcounterexamples
rohan@Ubuntu-22:~/open-source/dragonfly/src/core/search$ bison parser.y -Wcounterexamples
parser.y: warning: 9 shift/reduce conflicts [-Wconflicts-sr]
parser.y: warning: shift/reduce conflict on token "int64" [-Wcounterexamples]
  Example: NOT_OP search_expr • "int64"
  Shift derivation
    search_expr
    ↳ 8: NOT_OP search_expr
                ↳ 6: search_expr search_expr
                                 ↳ 10: • "int64"
  Reduce derivation
    search_expr
    ↳ 6: search_expr               search_expr
         ↳ 8: NOT_OP search_expr • ↳ 10: "int64"
```
But still i have never worked in parsers so not sure what does this mean in detail, but what i can see if that the grammar is ambiguous and that is what bison is complaining about, giving counter examples showing 2 separate outcomes being possible

**Comment by romange** [comment]
@rohan-puri  please note that there is someone who is working on this issue already

**Comment by rohan-puri** [comment]
@romange okay thanks for the information. I will look into something else.

**Comment by dranikpg** [comment]
Fixed by #1838
